NotationV0/src/verify.ts

```diff
@@ -0,0 +1,105 @@
+import * as path from 'path';
+import * as taskLib from 'azure-pipelines-task-lib/task';
+
+import { NOTATION, NOTATION_BINARY, TRUST_STORE, X509 } from './lib/constants';
+
+import { getArtifactReferences } from './lib/variables';
+import { getConfigHome } from './lib/fs';
+import { notationRunner } from './lib/runner';
+
+export async function verify(): Promise<void> {
+    const artifactRefs = getArtifactReferences();
+    const trustPolicy = taskLib.getInput('trustPolicy', true) || '';
+    const trustStore = taskLib.getInput('trustStore', true) || '';
+    const allowReferrerAPI = taskLib.getBoolInput('allowReferrersAPI', false);
+    const debug = taskLib.getVariable('system.debug')
+
+    // config trust policy
+    await configTrustPolicy(trustPolicy);
+    // config trust store
+    taskLib.rmRF(path.join(getConfigHome(), NOTATION, TRUST_STORE));
+    await configTrustStore(trustStore);
+
+    let env = { ...process.env }
+    if (allowReferrerAPI) {
+        env["NOTATION_EXPERIMENTAL"] = "1";
+    }
+
+    // run notation verify for each artifact
+    await notationRunner(artifactRefs, async (artifactRef: string) => {
+        return taskLib.tool(NOTATION_BINARY)
+            .arg(['verify', artifactRef, '--verbose'])
+            .argIf(allowReferrerAPI, '--allow-referrers-api')
+            .argIf(debug && debug.toLowerCase() === 'true', '--debug')
+            .exec({ env: env });
+    })
+}
+
+async function configTrustPolicy(trustPolicy: string): Promise<void> {
+    // run notation command to install trust policy
+    let code = await taskLib.tool(NOTATION_BINARY)
+        .arg(['policy', 'import', '--force', trustPolicy])
+        .exec();
+    if (code !== 0) {
+        throw new Error(`Failed to import trust policy: ${trustPolicy}`);
+    }
+
+    code = await taskLib.tool(NOTATION_BINARY)
+        .arg(['policy', 'show'])
+        .exec();
+    if (code !== 0) {
+        throw new Error(`Failed to show trust policy`);
+    }
+}
+
+// configTrustStore configures Notation trust store based on specs.
+// Reference: https://github.com/notaryproject/specifications/blob/v1.0.0-rc.2/specs/trust-store-trust-policy.md#trust-store
+async function configTrustStore(dir: string): Promise<void> {
+    const trustStoreX509 = path.join(dir, X509); // .github/truststore/x509
+    if (!taskLib.exist(trustStoreX509)) {
```

Comment:
Same here. We need to make sure this check is sync.

Reply:
Yes, it is sync.